### PR TITLE
samples: lwm2m carrier certicate handling updated

### PR DESCRIPTION
**Pull request by peho-nordic**
This commit aligns samples to follow new  LwM2M
CA certificate handling strategy and  application
needs to initilize needed tags and provide info
back to LwM2M carrier when event is sent.

### CODE REVIEW

**Comment by nordic-hani** [comment]
Looks good to me. Matches our internal changes.

- PR should be directed at libcarrier branch (not master).
    - libcarrier branch should be rebased on master to make sure changes are applied on up to date master.

Doc comments (This can be separate PR if we want):
- Add change to the changelog.
- Document the event in the documentation.
- Mention the use of certificate in the sample/nrf9160/lw2m2_carrier documentation? Not sure if that is needed, maybe a description in bin\lib\lwm2m_carrier\doc is sufficient.
- Is a comment needed in asset tracker documentation? Probably not?

**Comment by nordic-hani** [comment]
Issue with coloring and issue with new lines.

image

**Review by nordic-hani — APPROVED** [review]

File: applications/asset_tracker/src/carrier_certs/carrier_certs.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+#include <sys/printk.h>
+#include <modem/modem_key_mgmt.h>
+#include "carrier_certs.h"
+
+#define ORANGE "\e[0;33m"
+#define GREEN "\e[0;32m"
+#define NORMAL "\e[0m"
+
+/* The order of the tags in this array specifies which tag is used first */
+static int tags[] = { 411, 412 };
```
Comment (stig-bjorlykke) [inline applications/asset_tracker/src/carrier_certs/carrier_certs.c:20]: Do we document the sec_tags in use? We already use 25-28, and before this PR we used 11.

File: applications/asset_tracker/src/carrier_certs/carrier_certs.c [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+static const char ca411[] = {
+	/* VzW and Motive, AT&T */
```
Comment (stig-bjorlykke) [inline applications/asset_tracker/src/carrier_certs/carrier_certs.c:23]: Do we need to name-drop which operators are using which certificates here?

**Review by lemrey — APPROVED** [review]

**Comment by stig-bjorlykke** [comment]
Can we put the common code in `carrier_certs.c` and the certificates in one place?

File: applications/asset_tracker/src/carrier_certs/carrier_certs.c [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+static const ca_cert_tags_t ca_tags = {
+	.tags = tags,
+	.count = ARRAY_SIZE(tags),
+};
+
+int carrier_cert_provision(ca_cert_tags_t * const tags)
+{
+	int err;
+	bool mismatch = 0;
```
Comment (stig-bjorlykke) [inline applications/asset_tracker/src/carrier_certs/carrier_certs.c:58]: `bool mismatch = false`

File: applications/asset_tracker/src/carrier_certs/carrier_certs.h [hunk 4]
```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef CARRIER_CERTS_H_
+#define CARRIER_CERTS_H_
```
Comment (stig-bjorlykke) [inline applications/asset_tracker/src/carrier_certs/carrier_certs.h:8]: This header file is not equal for all copies, should be synced.
The guard name should be `CARRIER_CERTS_H__`.

**Review by umapraseeda — APPROVED** [review]

File: applications/asset_tracker/src/carrier_certs/carrier_certs.h (hunk 4 above)
Comment (peho-nordic) [reply applications/asset_tracker/src/carrier_certs/carrier_certs.h:8]: `asset_tracker` has both styles: some have `_H__` and some `_H_` suffix...
Will fix anyway.

Comment (peho-nordic) [reply applications/asset_tracker/src/carrier_certs/carrier_certs.h]: fixed

File: applications/asset_tracker/src/carrier_certs/carrier_certs.c (hunk 3 above)
Comment (peho-nordic) [reply applications/asset_tracker/src/carrier_certs/carrier_certs.c]: fixed

File: applications/asset_tracker/src/carrier_certs/carrier_certs.c (hunk 2 above)
Comment (peho-nordic) [reply applications/asset_tracker/src/carrier_certs/carrier_certs.c]: removed.

**Pull request merged**